### ISSUE TITLE
Better error message 

### ISSUE THREAD

**Issue by dinh** [TekWizely/run#32]
A script with an error at line 9
```
##
# Invokes hello
# EXPORT RUN := ${.RUN}
# EXPORT RUNFILE := ${.RUNFILE}
test:
# syntaxe error here <------
    "${RUN}" -r "${RUNFILE}" hello

hello:
    echo "Hello, World"
```

sh-4.4$ run test

Give this hard to understand message 
`run: non-EOF error returned from lexer, treating as EOF: 10:6: unexpected rune '"'`

which is not easy to understand

**Comment by TekWizely** [comment]
Hi @dinh ,

I would have preferred to see your original script that caused the error, but I'm guessing it also contained a comment in the same place as the `# syntax error` comment in your script.

So the cause of the error message was:
* The comment at the beginning of the line terminated the command script
* Run assumed the `"${RUN}"` line was a Run statement 

In other words, here is what Run sees:
```
##
# Invokes hello
# EXPORT RUN := ${.RUN}
# EXPORT RUNFILE := ${.RUNFILE}
test:
# <-- Any text at beginning of line terminates command script

# Run thinks this is a run statement, not a part of the 'test' script
"${RUN}" -r "${RUNFILE}" hello

hello:
    echo "Hello, World"
```
This is easily fixed by indenting your comment, but I can think of 3 enhancements that might be useful:

1) A better error message when a Run statement is expected and not found.
2) An error if a command contains an empty script - This would likely only help when the user terminates the command on the first line, as you did, but it seams reasonable that a command should have a non-empty script in general.
3) An error if there is leading whitespace when expecting a Run statement.  Chances are that an indented statement was meant to be part of a command script.

I'll probably create separate tickets for 2 and 3 above, but I'll use this ticket to track the change for 1.

Thanks for taking the time to report an issue !

-TW

**Comment by TekWizely** [comment]
@dinh I released v0.7.1 which now generates an error when a command script is empty.

It also contains a few changes that might hopefully create better error messages.

I'm going to close this ticket but please feel free to open another one if you run into any issues.

Thanks!
